source/src/BNFC/Backend/Haskell.hs

```diff
@@ -262,7 +262,7 @@ makefile opts cf makeFile = vcat
 
   -- | Rule to build Haskell test parser.
   testParserRule :: Doc
-  testParserRule = Makefile.mkRule tgt deps [ "${GHC} ${GHC_OPTS} $@" ]
+  testParserRule = Makefile.mkRule tgt deps [ "${GHC} -lpthread ${GHC_OPTS} $@" ]
```

Comment:
This way, the flag `-lpthread` cannot be omitted or overwritten from outside.
Better is to change the default of `GHC_OPTS` to `GHC_OPTS=-lpthread` in the generated makefile.

Reply:
Addressed.